contracts/mocks/Cover/CoverMockStakingPool.sol

```diff
@@ -75,7 +75,13 @@ contract CoverMockStakingPool is IStakingPool {
       ? calculateFixedPricePremium(amount, request.period, mockPrices[request.productId])
       : calculatePremium(mockPrices[request.productId], amount, request.period);
 
-    return (premium, allocationId);
+    if (amount == 0) {
+      return (0, 0);
+    }
+    uint extraPremium = premium * Math.max(
+      (amount - previousAllocationAmountInNXMRepriced), 0) / amount;
```

Comment:
This computation won't work correctly if amount <  previousAllocationAmountInNXMRepriced since we're using uint, need to compare the 2 values